### PR TITLE
Reactify microcerts header

### PR DESCRIPTION
**Pull request by mrgnr**
## Done

- Converted the header on the microcerts page to render with React

## QA

- Run the server with `dotrun`
- Go to http://localhost:8001/cube/microcerts?test_backend=true
- Verify that the header is displayed and corresponds to the badges you've received

## Issue / Card

Fixes https://github.com/canonical-web-and-design/commercial-squad/issues/369

### In progress
microcerts-header

### Complete
microcerts-header-complete

### CODE REVIEW

**Comment by wgx** [comment]
We discussed the speed issue:

- Display the header with placeholder text
- Display a loading spinner
- Replace the content when it loads

File: static/js/src/cube/components/CubeHeader/Hero.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+          setHeroStripIsShallow(true);
+          setHeroContentMarginTop("-2rem");
+        }
+      });
+    });
+
+    if (heroStrip.current) {
+      resizeObserver.observe(heroStrip.current);
+    }
+  }, [heroStrip.current]);
```
Comment (ClementChaumel) [inline static/js/src/cube/components/CubeHeader/Hero.tsx:39]: What does this do? Could this be done via CSS media queries?

File: static/js/src/cube/components/CubeHeader/CubeHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      {(isLoading && (
+        <section
+          className="p-strip--light is-slanted--top-right u-vertically-center u-align--center"
+          style={{ minHeight: "368px" }}
+        >
+          <i className="p-icon--spinner u-animation--spin u-align--center"></i>
+        </section>
+      )) ||
+        (certifiedBadge && <Certified certifiedBadge={certifiedBadge} />) ||
+        (hasEnrollments && <Prepare />)}
```
Comment (ClementChaumel) [inline static/js/src/cube/components/CubeHeader/CubeHeader.tsx:37]: Could you use the ternary syntax here instead? 
eg.
```jsx
{isLoading ? (
    <section>
        ...
    </section>
) : (
         ...
)}
```

File: static/js/src/cube/components/CubeHeader/Hero.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    0
+  );
+
+  const heading = isLoading
+    ? ""
+    : certifiedBadge
+    ? "CUBE 2020 complete"
+    : hasEnrollments
+    ? "Your road to CUBE"
+    : "The road to CUBE";
```
Comment (ClementChaumel) [inline static/js/src/cube/components/CubeHeader/Hero.tsx:52]: I know I just asked you to use this syntax in the render function but don't think it scales well with nested conditions. Here I find it hard to parse what's going on

File: static/js/src/cube/components/CubeHeader/Hero.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    ? ""
+    : certifiedBadge
+    ? `You have completed all ${modules.length} microcerts, proving your Ubuntu expertise. You are now a Certified Ubuntu Engineer.`
+    : hasEnrollments
+    ? passedCourses > 0
+      ? `You have completed ${passedCourses} microcerts.\nComplete ${
+          modules.length - passedCourses
+        } more microcerts to attain CUBE.`
+      : `Complete ${modules.length} microcerts to attain CUBE.`
+    : `The path to certification is clear and convenient. Complete ${modules.length} microcerts to attain CUBE.`;
```
Comment (ClementChaumel) [inline static/js/src/cube/components/CubeHeader/Hero.tsx:64]: same as above.

One idea could be to define heading and descriptions as empty variables and then do something like

```jsx
if(certifiedBadge){
  description = "..."
  heading = "Cube 2020 complete"
} else if(hasEnrollments){
  heading = "Your road to CUBE"
 ...
} else {
 ...
}
```

what do you think?

File: static/js/src/cube/header-app.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+const HeaderApp = () => {
+  return (
+    <QueryClientProvider client={queryClient}>
+      <App />
+    </QueryClientProvider>
+  );
+};
+
+export default HeaderApp;
```
Comment (ClementChaumel) [inline static/js/src/cube/header-app.tsx:45]: Wait, are we creating a new App just for the header? Couldn't this live in the same one as the microcerts table?

File: static/js/src/cube/components/CubeHeader/Hero.tsx (hunk 1 above)
Comment (mrgnr) [reply static/js/src/cube/components/CubeHeader/Hero.tsx:39]: This prevents the hero section from getting vertically squashed on smaller screens.

Setting the margin could definitely be done with a media query, but I'm not sure about applying the `is-shallow` class. I'll try to replace with a media query.

For reference, [the old code](https://github.com/canonical-web-and-design/ubuntu.com/pull/10953/files#diff-8e9f6cac8e2dd10b76e913ecb65ad88adbad037ec9d084a27cf3e506d53f8e56L215) did this with some javascript in the template.

Comment (mrgnr) [reply static/js/src/cube/components/CubeHeader/Hero.tsx]: I changed this to use CSS with a media query.

File: static/js/src/cube/components/CubeHeader/CubeHeader.tsx (hunk 2 above)
Comment (mrgnr) [reply static/js/src/cube/components/CubeHeader/CubeHeader.tsx]: Done :+1:

File: static/js/src/cube/components/CubeHeader/Hero.tsx (hunk 3 above)
Comment (mrgnr) [reply static/js/src/cube/components/CubeHeader/Hero.tsx]: I agree, fixed :+1:

File: static/js/src/cube/components/CubeHeader/Hero.tsx (hunk 4 above)
Comment (mrgnr) [reply static/js/src/cube/components/CubeHeader/Hero.tsx]: Fixed :+1:

File: static/js/src/cube/header-app.tsx (hunk 5 above)
Comment (mrgnr) [reply static/js/src/cube/header-app.tsx:45]: The template has different divs where the header and table are rendered, which is why they're two separate apps right now. When we reactify the rest of the page in https://github.com/canonical-web-and-design/commercial-squad/issues/424 everything can be under a single app.

File: static/sass/_pattern_cube-hero.scss [hunk 6]
```diff
@@ -0,0 +1,16 @@
+@mixin ubuntu-p-cube-hero {
+  .p-cube-hero {
+    .p-cube-hero--content {
+      margin-top: 2rem;
+    }
+
+    @media (min-width: $breakpoint-large) {
+      padding-bottom: 1.5rem !important;
```
Comment (ClementChaumel) [inline static/sass/_pattern_cube-hero.scss:8]: :scream: Could this be achieved with a higher specificity instead of `!important`?

**Review by ClementChaumel — APPROVED** [review]
Looks good! Can't wait for the two apps to be merged into one, it'll remove a lot of duplicate code.
(small CSS nitpick, feel free to ignore)

File: static/sass/_pattern_cube-hero.scss (hunk 6 above)
Comment (mrgnr) [reply static/sass/_pattern_cube-hero.scss]: I knew it felt dirty using `!important` :laughing: 
I removed it and increased the specificity by making the selector `.p-strip.p-cube-hero`.
